ci-operator/step-registry/openshift-power-monitoring/tests/kepler/openshift-power-monitoring-tests-kepler-commands.sh

```diff
@@ -2,41 +2,22 @@
 
 set -eu -o pipefail
 
+declare -r LOGS_DIR="/tmp/test-run-logs"
+
 must_gather() {
     oc get pods -n openshift-operators
     oc describe deployment kepler-operator-controller-manager -n openshift-operators
     oc logs -n openshift-operators deployments/kepler-operator-controller-manager
-    oc get pods -n openshift-kepler-operator
-    oc describe daemonsets/kepler-exporter-ds -n openshift-kepler-operator
-    oc logs -n daemonsets/kepler-exporter-ds -n openshift-kepler-operator
-}
-validate_kepler() {
-    local ret=0
-    echo "Validating Kepler"
-    oc get kepler kepler || {
-        echo "Missing Kepler Instance. Something wrong with Kepler deployment"
-        ret=1
-    }
-    oc rollout status -n openshift-kepler-operator daemonset kepler-exporter-ds || {
-        echo "Daemonset not in healthy state"
-        ret=1
-    }
-    return $ret
 }
+
 main() {
-    # below validation steps will be replaced by go tests in future
-    validate_kepler || {
-        must_gather
-        return 1
-    }
-    sleep 60
-    oc expose service/kepler-exporter-svc -n openshift-kepler-operator --name kepler-metrics-route
-    url=http://$(oc get route kepler-metrics-route -n openshift-kepler-operator -o jsonpath='{.spec.host}')/metrics
-    [[ $(curl "$url" | grep -c kepler_) -gt 0 ]] || {
-        echo "Kepler validation failed"
+
+    mkdir -p $LOGS_DIR
+    ./e2e.test -test.v -test.failfast 2>&1 | tee "$LOGS_DIR/e2e.log" || {
```

Comment:
could you please help me undestand why the test harness image needed if we are executing the tests from a compiled binary? 🤔 
Also where /how is the binary compiled?

Reply:
Well as per my understanding in ProwCI if you need to perform any operations it all happens inside a pod and since we don't have access to the repository inside the pod when triggering e2e tests that's why we specifically build an image that contains a compiled binary of tests.

Reply:
[Here](https://github.com/openshift/release/blob/f4fc15e582cdd5c6ba02894baa722e264a2324bd/ci-operator/config/openshift-power-monitoring/kepler-operator/openshift-power-monitoring-kepler-operator-v1alpha1.yaml#L26) our test image is built which is part of CI pipeline that uses Dockerfile present in our [repo](https://github.com/openshift-power-monitoring/kepler-operator/blob/v1alpha1/tests/Dockerfile)

Reply:
@vprashar2929 Thanks again but I think I will need to understand this better. Perhaps on a call :) 
I am going to lgtm so that the PR can go ahead

Reply:
Sure absolutely, Thanks